pkg/egressservice/egressservice.go

```diff
@@ -0,0 +1,394 @@
+		context.TODO(), 5*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
+			err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+				Name:      builder.Definition.Name,
+				Namespace: builder.Definition.Namespace,
+			}, egrSvc)
+
+			if err != nil && k8serrors.IsNotFound(err) {
+				glog.V(100).Infof("EgressService not found: %v", err)
+
+				return false, err
+			}
+
+			if err != nil && !k8serrors.IsNotFound(err) {
+				glog.V(100).Infof("Error retrieving EgressService: %v", err)
+
+				return false, nil
+			}
+
+			return err == nil && egrSvc != nil, err
+		})
+
+	return egrSvc, err
+}
+
+// Create makes a EgressService in the cluster and stores the created object in struct.
+func (builder *EgressServiceBuilder) Create() (*EgressServiceBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the EgressServcice %q in namespace %q",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+
+	if !builder.Exists() {
+		err = wait.PollUntilContextTimeout(
+			context.TODO(), 5*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
+				err = builder.apiClient.Create(context.TODO(), builder.Definition)
+
+				if err == nil {
+					glog.V(100).Infof("Created EgressServcice %q in namespace %q",
+						builder.Definition.Name, builder.Definition.Namespace)
+
+					builder.Object = builder.Definition
+
+					return true, nil
+				}
+
+				glog.V(100).Infof("Error creating EgressServcice: %v", err)
+
+				return false, nil
+			})
+	}
+
+	return builder, err
+}
+
+// Delete removes EgressService from a cluster.
+func (builder *EgressServiceBuilder) Delete() (*EgressServiceBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting EgressService %q in namespace %q",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("EgressService %q in namespace %q does not exist",
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		builder.Object = nil
+
+		return builder, nil
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		glog.V(100).Infof("Error deleting EgressService: %v", err)
+
+		return builder, fmt.Errorf("failed to delete EgressService due to %w", err)
+	}
+
+	glog.V(100).Infof("Deleted EgressService %q in %q namespace",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Update update EgressService object on cluster with content in the builder.
+func (builder *EgressServiceBuilder) Update() (*EgressServiceBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating EgressService %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := wait.PollUntilContextTimeout(
```

Comment:
Same comment about polling in functions, if we're doing polling in eco-goinfra it should be in a specific `WaitUntil***` function.

In update specifically, if the resource has been updated on the cluster since last `Pull` then the update will fail but this function will return `context.DeadlineExceeded` instead of the actual error

Reply:
:sigh: removed